docs/_docs/persistence/snapshot.adoc

```diff
@@ -0,0 +1,188 @@
+* snapshot may be restored only at the same baseline topology with the same node consistent IDs;
+
+== Configuration
+
+The destination snapshot directory can be configured via `IgniteConfiguration`.
+
+[tabs]
+--
+tab:XML[]
+
+[source, xml]
+----
+include::code-snippets/xml/snapshots.xml[tags=ignite-config;!discovery, indent=0]
+
+----
+
+tab:Java[]
+
+[source, java]
+----
+include::{javaCodeDir}/Snapshots.java[tags=config, indent=0]
+
+----
+
+tab:C#/.NET[]
+tab:C++[]
+--
+
+== Snapshot creation
+
+Ignite provides the ability to start a snapshot operation from the following interfaces:
+
+- link:#command-line[Command line]
+- link:#jmx[JMX]
+- link:#java-api[Java API]
+
+=== Command line
+
+Ignite ships `control.(sh|bat)` scripts, located in the `$IGNITE_HOME/bin` directory, that act like a tool to
+start or cancel snapshot operation from the command line. The following commands can be used with `control.(sh|bat)`:
+
+[source,shell]
+----
+#Create cluster snapshot:
+control.(sh|bat) --snapshot create snapshot_name
+
+#Cancel running snapshot:
+control.(sh|bat) --snapshot cancel snapshot_name
+
+#Kill running snapshot:
+control.(sh|bat) --kill SNAPSHOT snapshot_name
+----
+
+=== JMX
+
+You can start snapshot operation or cancel currently running one via the `SnapshotMXBean` interface:
+
+[cols="1,1",opts="header"]
+|===
+|Method | Description
+|createSnapshot(String snpName) | Create cluster-wide snapshot.
+|cancelSnapshot(String snpName) | Cancel started cluster-wide snapshot on the node initiator.
+|===
+
+=== Java API
+
+The snapshot operation can be started using Java API:
+
+[tabs]
+--
+tab:Java[]
+
+[source, java]
+----
+include::{javaCodeDir}/Snapshots.java[tags=create, indent=0]
+----
+--
+
+== Restoring from a snapshot (manual)
+
+NOTE: Removing data from the $IGNITE_HOME/work directory performed at your own risk.
+
+Currently, the data restore procedure is fully manual. Since the snapshot is a consistent copy of all data files for each node,
+in order to recover from the particular snapshot, you need to start each node of the same cluster on these files
+(see link:#snapshot-structure[Snapshot Structure] for details).
+
+To restore a cluster from a snapshot, the following steps should be performed for each baseline node in it:
+
+- Locate the snapshot you've created by name;
+- Remove all the data from `$IGNITE_HOME/work/{node_id}` where `{node_id}` is a consistent id of a node you work with;
+- Copy all snapshot files from the snapshot related to `{node_id}` to the $IGNITE_HOME/work/{node_id} directory;
+
+Some use cases at production deployments may require creating a snapshot at the m-node cluster and applying it to the
+n-node cluster. Here are some options:
+
+[cols="1,1",opts="header"]
+|===
+|Condition | Description
+|m == n | The *recommended* case. Use the same baseline and configuration.
+|m < n | Start the m-node cluster from the snapshot and add additional nodes to the started baseline. The process
```

Comment:
Let's say I had 5 nodes in the N-cluster and trying to apply the snapshot to a 3-node cluster (the m-cluster). I will copy the snapshot files from the first 3 nodes of the N-cluster to the M-cluster.  But what should I do with the snapshot files of the other 2 nodes of the N-cluster? Basically, what's the procedure for squeezing 5 folders (N-cluster) with the snapshots into the 3 folders (M-cluster). That's unclear to me.

Reply:
Denis, 

- the m-cluster is the source of a snapshot (take from)
- the n-cluster is the destination (apply to)

case `m > n` is not supported yet, since, as you mentioned before, `squeezing` procedures are not developed yet.
case `m < n` start m-cluster from a snapshot and add additional nodes to it by changing baseline.